src/kv/untyped_map.h

```diff
@@ -626,34 +605,27 @@ namespace kv::untyped
 
     std::unique_ptr<AbstractMap::Snapshot> snapshot(Version v) override
     {
+      // This takes a snapshot of the state of the map at the last entry
+      // committed at or before this version. The Map expects to be locked while
+      // taking the snapshot.
       auto r = roll.commits->get_head();
-      while (r != nullptr)
+
+      for (auto current = roll.commits->get_tail(); current != nullptr;
```

Comment:
Simplifying this loop here which had the same logic as the one in `create_view()` (i.e. find the commit with the largest value `<=` the target `v`).